### PR TITLE
feat:H1099-L3 : Refund creation and sync for Square

### PR DESCRIPTION
**Pull request by BhardwajShrey**
## Type of Change

- [x] New feature

## Description
Refund creation and sync for Square

### Additional Changes

- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/square/transformers.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            },
+            idempotency_key: Uuid::new_v4().to_string(),
+            source_id: match &item.request.statement_descriptor_suffix {
+                Some (val) => String::from(val),
+                None => String::from("cnon:card-nonce-ok")
+            }
+        };
+        Ok(payment_request)
+    }
+}
```
Comment (Narayanbhat166) [inline crates/router/src/connector/square/transformers.rs:36]: @bernard-eugine Have a look at this. I don't think we can use that field for this purpose

File: crates/router/src/services/api.rs [hunk 2]
```diff
@@ -163,7 +163,9 @@ where
         payments::CallConnectorAction::Trigger => {
             match connector_integration.build_request(req, &state.conf.connectors)? {
                 Some(request) => {
+                    println!("ReqIsHere {:#?}", request);
```
Comment (Narayanbhat166) [inline crates/router/src/services/api.rs:166]: remove these

**Review by Narayanbhat166 — CHANGES_REQUESTED** [review]
Remove debugging println statements, and format the code

File: crates/router/src/connector/square/transformers.rs (hunk 1 above)
Comment (bernard-eugine) [reply crates/router/src/connector/square/transformers.rs:36]: Are we referring to source_id field? Why us "cnon:card-nonce-ok" being passed for None case?

**Comment by ArjunKarthik** [comment]
Thanks for the contribution @BhardwajShrey , We are merging the changes from different participant as it has more flows implemented. You can still continue your contributions on the [open issues](https://github.com/juspay/hyperswitch/issues?q=is%3Aissue+is%3Aopen+label%3AA-connector-integration).

**Pull request closed without merging**